### PR TITLE
[W5.4][F11-A1]Se Hyun Park

### PR DESCRIPTION
**Pull request by sofarsophie**
Added an 'Affiliations' field by modelling it on the pre-existing 'Tags' field. Added necessary classes and modified files/methods/tests etc. 'Affiliations' field allows you to add your contacts' official schools and/or office (ex. NUS, Google).

Please help review the PR! :)

### CODE REVIEW

**Comment by karrui** [comment]
Seems like you forgot to update your expected.txt when you fixed the typo!

**Comment by karrui** [comment]
Overall, seems like lots of reuse from the relevant tag classes as it is also a form of tags. Would it be better if Affiliation inherited from the tag class instead?

**Comment by sofarsophie** [comment]
@karrui Yeah, quite true! When I first decided to add the Affiliations, I planned them as being quite different from Tags (for ex. Affiliations can be used to build a 'Network' in the AddressBook). But the implementation was pretty much the same as Tags like you said! 

+ will update the expected.txt! :)

File: test/java/seedu/addressbook/data/AddressBookTest.java [hunk 1]
```diff
@@ -28,6 +30,11 @@
     private Tag tagMathematician;
     private Tag tagEconomist;
 
+    private Affiliation affiliationNUS;
+    private Affiliation affiliationGoogle;
+    private Affiliation affiliationIBM;
+    private Affiliation affiliationCMU;
```
Comment (PierceAndy) [inline test/java/seedu/addressbook/data/AddressBookTest.java]: Abbreviations and acronyms should not be uppercase when used as a part of a name.
See point 6 here: https://github.com/oss-generic/process/blob/master/codingStandards/CodingStandard-Java.adoc#naming-conventions

File: test/java/seedu/addressbook/storage/StorageFileTest.java [hunk 2]
```diff
@@ -11,14 +11,17 @@
...
 import seedu.addressbook.data.exception.IllegalValueException;
 import seedu.addressbook.data.person.Address;
 import seedu.addressbook.data.person.Email;
 import seedu.addressbook.data.person.Name;
 import seedu.addressbook.data.person.Person;
 import seedu.addressbook.data.person.Phone;
 import seedu.addressbook.data.tag.Tag;
+import seedu.addressbook.data.affiliation.Affiliation;
 import seedu.addressbook.data.tag.UniqueTagList;
+import seedu.addressbook.data.affiliation.UniqueAffiliationList;
```
Comment (PierceAndy) [inline test/java/seedu/addressbook/storage/StorageFileTest.java]: Duplicate import statement

File: test/java/seedu/addressbook/storage/StorageFileTest.java [hunk 3]
```diff
@@ -11,14 +11,17 @@
...
 import seedu.addressbook.data.AddressBook;
+import seedu.addressbook.data.affiliation.UniqueAffiliationList;
 import seedu.addressbook.data.exception.IllegalValueException;
 import seedu.addressbook.data.person.Address;
 import seedu.addressbook.data.person.Email;
 import seedu.addressbook.data.person.Name;
 import seedu.addressbook.data.person.Person;
 import seedu.addressbook.data.person.Phone;
 import seedu.addressbook.data.tag.Tag;
+import seedu.addressbook.data.affiliation.Affiliation;
```
Comment (PierceAndy) [inline test/java/seedu/addressbook/storage/StorageFileTest.java]: Unused import statement

File: test/java/seedu/addressbook/util/TestUtil.java [hunk 4]
```diff
@@ -27,6 +28,8 @@
 import seedu.addressbook.data.person.UniquePersonList.DuplicatePersonException;
 import seedu.addressbook.data.tag.Tag;
 import seedu.addressbook.data.tag.UniqueTagList;
+import seedu.addressbook.data.affiliation.Affiliation;
```
Comment (PierceAndy) [inline test/java/seedu/addressbook/util/TestUtil.java]: Unused import statement

File: test/java/seedu/addressbook/util/TestUtil.java [hunk 5]
```diff
@@ -27,6 +28,8 @@
...
+import seedu.addressbook.data.affiliation.UniqueAffiliationList;
```
Comment (PierceAndy) [inline test/java/seedu/addressbook/util/TestUtil.java]: Duplicate import statement

File: test/java/seedu/addressbook/util/TestUtil.java [hunk 6]
```diff
@@ -111,7 +114,7 @@ public static AddressBook clone(AddressBook addressBook) {
     public static Person generateTestPerson() {
         try {
             return new Person(new Name(Name.EXAMPLE), new Phone(Phone.EXAMPLE, false),
-                    new Email(Email.EXAMPLE, true), new Address(Address.EXAMPLE, false), new UniqueTagList());
+                    new Email(Email.EXAMPLE, true), new Address(Address.EXAMPLE, false), new UniqueTagList(), new UniqueAffiliationList());
```
Comment (PierceAndy) [inline test/java/seedu/addressbook/util/TestUtil.java]: Keep lines no longer than 120 chars.
See points 2 & 3 here: https://github.com/oss-generic/process/blob/master/codingStandards/CodingStandard-Java.adoc#layout

**Comment by PierceAndy** [comment]
@sofarsophie Some comments added. Please close the PR after reading comments.

**Pull request closed without merging**